src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +27,46 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         }
         this.value = address;
     }
+    
+    public void addressSplit(String address) {
+    	int startPosition = 0;
+    	Block block = new Block();
+    	Street street = new Street();
+    	Unit unit = new Unit();
+    	PostalCode postalCode = new PostalCode();
+    	
+    	for (int i=0; i<address.length(); i++) {
+    		if (address.charAt(i) == ',') {
+    			block.setBlock(address.substring(startPosition, i));
+    			startPosition = i + 2; // To eat the white space
+    			break;
+    		}
+    	}
+    	
+    	for (int i=startPosition; i<address.length(); i++) {
+    		if (address.charAt(i) == ',') { 
+    			street.setStreet(address.substring(startPosition, i));
+    			startPosition = i + 2;
+    			break;
+    		}
+    	}
+    	
+    	for (int i=startPosition; i<address.length(); i++) {
+    		if (address.charAt(i) == ',') { 
+    			unit.setUnit(address.substring(startPosition, i));
+    			startPosition = i + 2;
+    			break;
+    		}
+    	}
+    	
+    	for (int i=startPosition; i<address.length(); i++) {
+    		if (address.charAt(i) == ',') { 
+    			postalCode.setPostalCode(address.substring(startPosition, i));
+    			startPosition = i + 2;
+    			break;
+    		}
```

Comment:
Indentation problem here? Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.